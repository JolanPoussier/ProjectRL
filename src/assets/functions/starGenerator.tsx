import Image from "next/image";
import starImage from "src/assets/star/star.png"; 


export default function starGenerator ({star}: {star: number}) {
    const starsArray = [];

    for (let index = 0; index < star; index++) {
        starsArray.push(<Image src={starImage} alt="Etoile" width={30} height={30}/>)
    }

    return starsArray;
}