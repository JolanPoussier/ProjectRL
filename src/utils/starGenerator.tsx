import Image from "next/image";
import starImage from "src/assets/star/star.png"; 


export default function starGenerator ({star}: {star: number}) {
    const starsArray = [];

    for (let index = 0; index < star; index++) {
        starsArray.push(<Image key={index} src={starImage} alt="Etoile" width={30} height={30}/>)
    }

    return starsArray;
}