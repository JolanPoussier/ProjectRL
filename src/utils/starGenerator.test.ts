import starGenerator from './starGenerator'

// Configure Jest Snapshot Testing
expect.addSnapshotSerializer({
  print(val) {
    return String(val)
  },
  test(val) {
    return typeof val === 'string'
  },
})

describe('starGenerator', () => {
  it('renders stars correctly', () => {
    const stars = starGenerator({ star: 3 })
    const starOutput = stars.map(starElement => (starElement.key ? starElement.key.toString() : null))

    // Use Jest Snapshot Testing to capture and compare snapshots
    expect(starOutput).toMatchSnapshot()
  })

  it('generates no stars for a count of 0', () => {
    const starCount = 0
    const stars = starGenerator({ star: starCount })

    // Assert that no stars were generated
    expect(stars.length).toBe(0)
  })
})
