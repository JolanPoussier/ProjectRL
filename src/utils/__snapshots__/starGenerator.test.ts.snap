// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`starGenerator renders stars correctly 1`] = `
[
  0,
  1,
  2,
]
`;
